### PR TITLE
Stats: Add Jetpack Stats products supporting check

### LINKED ISSUE #78654: Paid Stats: Add products for Calypso
FG/store-gt-and-dev-resources/how-to-add-products-to-the-store

### PR DESCRIPTION
**Pull request by kangzj**
Related to #78654 

## Proposed Changes

The PR adds the ability to check whether a site has Stats product or a plan that supports Stats.

#### Selectors:
- `hasSiteProductJetpackStats`: returns true if the site has has Paid or Free products
- `hasSiteProductJetpackStatsPaid`: returns true if the site has Paid Stats products

The PR also introduces `FEATURE_JETPACK_STATS_PAID` and `FEATURE_JETPACK_STATS_FREE` as features, which would be used for plans that support Jetpack Stats Paid or Free.

## Testing Instructions

* Sandbox Store following instructions at pejTkB-FS-p2
* Purchase Stats for a site `https://wordpress.com/checkout/{site-slug}/jetpack_stats_monthly`
* Open Calypso Live
* Add a couple of lines somewhere, for example: https://github.com/Automattic/wp-calypso/blob/aa7dfcbfe0d3eb0ef5c6e2020bfd9fba68bacfff/client/my-sites/stats/site.jsx#L475

```JavaScript
		const hasJetpackStats = hasSiteProductJetpackStats( state );
		const hasJetpackStatsPaid = hasSiteProductJetpackStatsPaid( state );
		console.log( 'hasJetpackStats', hasJetpackStats );
		console.log( 'hasJetpackStatsPaid', hasJetpackStatsPaid );
```

image

- Ensure the output in console is expected for sites

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by kangzj** [comment]
I'll try to add some unit tests for the selectors tomorrow.

**Comment by kangzj** [comment]
Tests for selectors added 😄

**Review by grzegorz-cp — APPROVED** [review]
👍

File: packages/calypso-products/src/constants/features.ts [hunk 1]
```diff
@@ -180,6 +180,8 @@ export const FEATURE_JETPACK_ANTI_SPAM = PRODUCT_JETPACK_ANTI_SPAM;
 export const FEATURE_JETPACK_ANTI_SPAM_MONTHLY = PRODUCT_JETPACK_ANTI_SPAM_MONTHLY;
 export const FEATURE_JETPACK_SEARCH = PRODUCT_JETPACK_SEARCH;
 export const FEATURE_JETPACK_SEARCH_MONTHLY = PRODUCT_JETPACK_SEARCH_MONTHLY;
+export const FEATURE_JETPACK_STATS_PAID = 'jetpack-stats-paid';
```
Comment (dognose24) [inline packages/calypso-products/src/constants/features.ts]: Should we aline features with products as [other plugins](https://github.com/Automattic/wp-calypso/blob/add/jetpack-stats-products/packages/calypso-products/src/constants/features.ts#L181-L182) do? We could iterate it when the plan support gets clearer.

**Review by dognose24 — APPROVED** [review]
These changes look reasonable and work well with the sandbox purchase flow. @kangzj Awesome work! 👍🏼

**Pull request merged**